impls/src/adapters/emoji.rs

```diff
@@ -1194,14 +372,73 @@ impl EmojiSlate {
 			emoji_map_idx.push(bv.load::<u16>());
 		}
 
+		// get the emoji string from slate_json
 		let emoji_string = self.translate2emoji(emoji_map_idx);
 
-		return emoji_string;
+		let final_emoji_string = match TYPE_TRANSACTION {
+			// if compressed
+			"compress" => {
+				// Get a default header (Gzip method and Version 1)
+				let header = Header::default();
+				// Turns the header into emoji_string
+				let h_emoji = header.to_emoji_string();
+				// Merge all strings of Emojis: Header + Divider + Transaction
+				let emoji_string_header = h_emoji + EMOJI_DIVIDER.glyph + &emoji_string;
+
+				emoji_string_header
+			}
+			// If normal method
+			_ => emoji_string,
+		};
+
+		return final_emoji_string;
 	}
 
+	/// Transform a emoji string into a Slate
 	pub fn decode(&self, emoji_string: &str) -> Result<Slate, Error> {
-		let slate_string = self.translate2string(emoji_string);
+		// get the Emoji Divider
+		let div = EMOJI_DIVIDER.glyph.to_string().chars().next().unwrap();
+		// If countain divider -> Compressed with header
+		let (header, compressed_msg) = if emoji_string.contains(div) {
+			// Split the emoji_string into divider emoji
+			let mut splited_slate = emoji_string.split(div);
+
+			// get the header emojis (before divider)
+			let header_emoji = splited_slate.next().unwrap();
+			// get the transaction emojis (after divider)
+			let slate_emoji = splited_slate.next().unwrap();
+
+			// get the compressed message from emoji string
+			let compressed_msg = self.translate(slate_emoji, TYPE_TRANSACTION);
 
+			// get the header from emoji string
+			let header = Header::to_header(header_emoji.to_string());
+
+			(header, compressed_msg)
+		} else {
+			// If you don't have the divider, we create an artificial header with the Gzip compression method and consequently the Version is 0
+			let header = Header::new(CompressionFormat::Gzip, 0);
+
+			// Get the compressed message from emoji string
+			let compressed_msg = self.translate(emoji_string, TYPE_TRANSACTION);
+
+			(header, compressed_msg)
+		};
+
+		// If the message is compressed
+		let slate_string = match compressed_msg {
+			TranslateType::Compressed(compressed_vec) => {
+				// get the method of compression from emoji message
+				let compress_method = header.algo.clone();
+				// get the number of version from emoji message (don't used yet)
+				let emoji_version = header.version.clone();
```

Comment:
Another warning in this line, use the underscore to fix it:
```suggestion
				let _emoji_version = header.version.clone();
```